### DISCUSSION TITLE
Requirements to try out Dev branch

### DISCUSSION THREAD

**Discussion by rleongcs** [Blackymas/NSPanel_HA_Blueprint#784] [Q&A] [answered]
Hi...I would love to try out the Dev branch. Unfortunately I am not sure what are the requirements so as not to mess up my production devices. Below is my setup. I can use 1 of the NSPanel for testing but it still need to be connected to the Prod HA to control the lights. Just not sure how I can have the different Blueprint/TFT/ESPHome coexits?

Current setup:
NSPanel - 3 EU units in use.
HA - Proxmox VM. Pretty up-to-date either one version before or latest.
Blueprint - Always the latest.
TFT - Always the latest.
ESPHome - Always the latest.

**Comment by edwardtfn** [comment, marked as answer]
Dev branch right now is pretty similar to main, as we had a release just a few days ago, but if you are using the EU version you will be able to see our latest features which makes quite faster to load the Home page.
However, it could be quite buggy some times, be aware of that.

In my case, I've manually created another file on the Blueprint folder to host the code from dev and then manually copy/paste the code from GitHub.
Then in one of those I've slightly changed the name. This is not required, but makes much easier selecting in the automation screen:
![IMG_6210](https://user-images.githubusercontent.com/94725493/236330802-bee2fc74-858e-4d33-bce8-c8e86a3d3969.jpeg)
![IMG_6213](https://user-images.githubusercontent.com/94725493/236330820-86208d29-81a0-4ee9-a0d9-b4f01f05a75c.jpeg)

From the ESPHome side, on the external componentes area, change `ref: main` to `ref: dev` and compile again.

I also have 3 panels, with 2 of those running `main` and the other one running `dev`.

**Reply by rleongcs** [reply]
I would also have to upload the TFT file is it has update right? I see the EU TFT is updated yesterday.

**Reply by edwardtfn** [reply]
That's right.

**Reply by edwardtfn** [reply]
By the way, I would love if you can report any issue you find in `dev`, so we avoid those getting to `main`. Just let us know ij which environment you find the issue. 😉

**Reply by rleongcs** [reply]
That's my purpose why I wanted to setup my dev NSPanel 😉

**Reply by rleongcs** [reply]
How do I know the list of features to test for in the Dev branch? Is there a list maintained somewhere?

**Reply by edwardtfn** [reply]
I'm not aware about such list, but you can always see the comparison between `dev` and `main`: https://github.com/Blackymas/NSPanel_HA_Blueprint/compare/main...dev

**Reply by Doughboy68** [reply]
> Dev branch right now is pretty similar to main, as we had a release just a few days ago, but if you are using the EU version you will be able to see our latest features which makes quite faster to load the Home page. However, it could be quite buggy some times, be aware of that.
> 
> In my case, I've manually created another file on the Blueprint folder to host the code from dev and then manually copy/paste the code from GitHub. Then in one of those I've slightly changed the name. This is not required, but makes much easier selecting in the automation screen: ![IMG_6210](https://user-images.githubusercontent.com/94725493/236330802-bee2fc74-858e-4d33-bce8-c8e86a3d3969.jpeg) ![IMG_6213](https://user-images.githubusercontent.com/94725493/236330820-86208d29-81a0-4ee9-a0d9-b4f01f05a75c.jpeg)
> 
> From the ESPHome side, on the external componentes area, change `ref: main` to `ref: dev` and compile again.
> 
> I also have 3 panels, with 2 of those running `main` and the other one running `dev`.

Not sure if this is right spot, but your post reminded me of an idea that maybe the Blueprint files should be named with the version number (eg. NSPanel Configuration 3.3). This way, multiple panels can be added/updated easier. Last time I added a new panel, there was significant changes and it broke all of them when I redid the blueprint file, until I upgraded tft and firmware on remaining ones.

**Reply by rleongcs** [reply]
Good idea if you are going to be testing out different versions of the Blueprint.

**Comment by rleongcs** [comment]
Great...I have my Dev branch NSPanel up and running and I definitely felt the speed! 😆

**Comment by edwardtfn** [comment]
https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/dev/docs/different_version.md
